src/typ.rs

```diff
@@ -38,6 +38,13 @@ pub enum Type {
 }
 
 impl Type {
+    pub fn is_result(&self) -> bool {
+        match self {
+            Type::App { name, .. } if &*name == "Result" => true,
```

Comment:
Is this the most idiomatic way of checking `String` equality? I was getting really mixed answers from my Googling

Reply:
The left is a String and the right is a &str.

`&*name` is a converstion from String to &str, so it's then &str == &str

I think you might not need to do this if you swap the left and right hand side? I can't remember how the Eq and PartialEq traits work...

I think the string types are badly named in Rust. It would be clearer if `str` was called `string` and `String` was called `StringBuf` (StringBuffer).

Reply:
Thanks for the explanation! Swapping the left and right side works too it seems